src/Logger.ts

```diff
@@ -0,0 +1,41 @@
+import { EventEmitter } from 'events';
+
+import Entry from './Entry';
+import TimeProvider from './TimeProvider';
+
+/**
+ * @author Maciej Chałapuk (maciej@chalapuk.pl)
+ */
+export class Logger<Level> extends EventEmitter {
+  constructor(
+    private getTimestamp : TimeProvider,
+    private name : string,
+  ) {
+    super();
+  }
+
+  log(level : Level, message : string, meta : Object = {}) {
```

Comment:
What about convenience debug/info/warn/error? Are we planning to have some default string substitution mechanism for the `message`?

Reply:
Yes, I will add `DefaultLevel` and `DefaultLogger` later. As for string substitution, I think it's not needed when we have ES6 template strings.